core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -251,12 +255,22 @@ object ConsoleProducer {
     val options = parser.parse(args : _*)
     if(args.length == 0)
       CommandLineUtils.printUsageAndDie(parser, "Read data from standard input and publish it to Kafka.")
-    CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
 
     val useOldProducer = options.has(useOldProducerOpt)
-    val topic = options.valueOf(topicOpt)
+    val bootstrapServer = options.valueOf(bootstrapServerOpt)
```

Comment:
It's to make it easier for users. They don't have to switch the CLI param to switch between new/old producer. After all `broker-list` is not the same as `metadata.broker.list` either.